libraries/AdaptiveExpressions/ExpressionFunctions.cs

```diff
@@ -4495,6 +4495,30 @@ private static IEnumerable<object> Flatten(IEnumerable<object> list, int dept)
                 new ExpressionEvaluator(ExpressionType.Coalesce, Apply(args => Coalesce(args.ToArray())), ReturnType.Object, ValidateAtLeastOne),
                 new ExpressionEvaluator(ExpressionType.XPath, ApplyWithError(args => XPath(args[0], args[1])), ReturnType.Object, (expr) => ValidateOrder(expr, null, ReturnType.Object, ReturnType.String)),
                 new ExpressionEvaluator(ExpressionType.JPath, ApplyWithError(args => JPath(args[0], args[1].ToString())), ReturnType.Object, (expr) => ValidateOrder(expr, null, ReturnType.Object, ReturnType.String)),
+                new ExpressionEvaluator(
```

Comment:
This method is 1878 lines long. 

Please refactor. 

Consider patterns, such as a decorator, visitor, or strategy pattern.

Reply:
@cosmicshuai let's put down a task for this, @chrimc62 do you have any suggestions for this?

I somehow think this more as a data table instead of a code. So, maybe we can make it more looks like a piece of data?

Reply:
This function is fine as it is.  It is a table that is initialized with all of the possible functions.  None of the patterns Chris mentions are relevant.  Code that does an explicit push or artificially breaks up into functions is not clearer and in fact obscures that this is in fact a big table.

Reply:
@boydc2014 I agree, we need to create a new task on refactor this function table. One thing I think we can do is we can create a timex transform method so that several timex functions can be written in a more concise way.

Reply:
If we want to improve this, an alternative might be to use classes instead of the lambdas everywhere.  That way each class can override what it needs to and the table literally becomes a new of every class.  This would be a big change though.

Reply:
I created a new task on this, so we could discuss whether it is necessary to do this or how can we do the refactor.